src/components/Pagination/Pagination.js

```diff
@@ -57,6 +64,12 @@ export default class Pagination extends Component {
 
   componentWillMount() {
     this.uniqueId = `${Math.floor(Math.random() * 0xffff)}`;
+    this.pageInputDebouncer = debounce(
```

Comment:
This code seems to require the scenario of re-mounting.

Reply:
Are you suggesting `flush` the debouncer when component unmounts?

Reply:
Checked `flush` - It doesn't seem something that cancels the debouncer. `cancel()` of the debouncer is much closer, which we can run at `cDU`.

Reply:
Do we need to create this class instance property in `cWM`? Especially since this method is deprecated in 16.3+. Seems like we could just have:

```js
onPageInput = debounce(page => {
  if (page > 0) {
    this.props.onChange({ page, pageSize: this.state.pageSize });
  }
}, this.props.onChangeInterval)
```

Reply:
Like @asudoh mentioned in other comment, we might need it to `cancel` debounce with `cWU`

Reply:
@ryshah am fine with having it be cancellable, the main issue is placing it in `cWM` since that lifecycle method is deprecated after 16.3.

Reply:
@joshblack I see your concern. After reading the documentation, looks like React recommends adding subscriptions in `cDM` method instead of `cWM`. I will make that change instead.